src/parser.ts

```diff
@@ -1678,14 +1678,16 @@ export class Parser {
             let token = startToken;
             expr = this.parseConditionalExpression();
 
-            if (token.type === Token.Identifier && (token.lineNumber === this.lookahead.lineNumber) && token.value === 'async' && (this.lookahead.type === Token.Identifier)) {
-                const arg = this.parsePrimaryExpression();
-                this.reinterpretExpressionAsPattern(arg);
-                expr = {
-                    type: ArrowParameterPlaceHolder,
-                    params: [arg],
-                    async: true
-                };
+            if (token.type === Token.Identifier && (token.lineNumber === this.lookahead.lineNumber) && token.value === 'async') {
```

Comment:
Unfortunately, `match` family is typically for the lookahead token, not the current token. Hopefully we can specialize such a distinctive "identifier" someday with another approach.